components/DiscussionsCard/DiscussionsCard.tsx

```diff
@@ -21,41 +41,121 @@ const DiscussionsCard: React.FC<DiscussionsCardProps> = ({
   userPic,
   timeStamp,
   content,
-  replyClick,
-  editClick,
   deleteClick
 }) => {
-  return (
-    <div className={isMain ? styles.card_body_main : styles.card_body_sub}>
-      <div
-        className={
-          isMain ? styles.card_user_info_main : styles.card_user_info_sub
-        }
-      >
-        <div>
-          <img src={userPic} className={styles.user_pic} />
-          <span className={styles.username}>{username}</span>
+  const [replyMode, setReplyMode] = useState(false)
+  const [editMode, setEditMode] = useState(false)
+  const [replyText, setReplyText] = useState('')
+  const [editText, setEditText] = useState('')
+
+  const replyClick = () => {
+    setReplyMode(true)
+  }
+  const cancelReply = () => {
+    setReplyMode(false)
+    setReplyText('')
+  }
+  const editClick = () => {
+    setEditMode(true)
+    setEditText(content)
+  }
+
+  const cancelEdit = () => {
+    setEditMode(false)
+    setEditText('')
+  }
+
+  const replyBox = //onclick function for send has yet to be implemented, will be in future PR
+    (
+      <div className={styles.reply_box}>
+        <MdInput value={replyText} onChange={setReplyText} />
+        <button
+          className={styles.cancel_reply}
+          onClick={cancelReply}
+          data-testid="cancel_reply"
+        >
+          <CircleSlashIcon size={15} />
           <span> </span>
-          <span> - </span>
-          <span>{timeStamp}</span>
+          Cancel
+        </button>
+        <button className={styles.send_reply}>
+          <PaperAirplaneIcon size={15} />
+          <span> </span>
+          Send
+        </button>
+      </div>
+    )
+  const editBox = //onclick function for save has yet to be implemented, will be in future PR
+    (
+      <div className={styles.edit_box}>
+        <MdInput value={editText} onChange={setEditText} />
+        <div className={styles.edit_buttons}>
+          <button
+            onClick={cancelEdit}
+            className={styles.discard_edit}
+            data-testid="cancel_edit"
+          >
+            <TrashIcon size={15} />
+            <span> </span>
+            Discard
+          </button>
+          <button className={styles.discard_save}>
+            <ChecklistIcon size={15} />
+            <span> </span>
+            Save
+          </button>
         </div>
       </div>
-      <div className={styles.card_content}>{content}</div>
-      <div className={styles.card_buttons}>
-        <div className={styles.buttons_left}>
-          <div className={styles.reply_button} onClick={replyClick}>
-            <CommentIcon /> Reply
+    )
+
+  return (
+    <div className={styles.main_container}>
+      <div className={isMain ? styles.card_body_main : styles.card_body_sub}>
+        <div
+          className={
+            isMain ? styles.card_user_info_main : styles.card_user_info_sub
+          }
+        >
+          <div>
+            <img src={userPic} className={styles.user_pic} />
```

Comment:
You might want to add `alt=""` here. (https://davidwalsh.name/accessibility-tip-empty-alt-attributes)

Reply:
We should use the `<Image>` component by next.js

Reply:
> We should use the `<Image>` component by next.js

When I use the Image component from next, for some reason it did not incorporate any of the styling, so I kept it as <img> for now.

Reply:
> > We should use the `<Image>` component by next.js
> 
> When I use the Image component from next, for some reason it did not incorporate any of the styling, so I kept it as `<img>` for now.

Does these help?

- https://nextjs.org/docs/api-reference/next/image#style
- https://www.peterlunch.com/snippets/next-image-styling